ts/data.ts

```diff
@@ -1,6 +1,22 @@
 const data = {
   view: 'entry-form',
-  entries: [],
+  entries: readData(),
   editing: null,
   nextEntryId: 1,
 };
+
+// local storage
+
+function writeData(): void {
+  const jsonData = JSON.stringify(data);
+  localStorage.setItem('entries-storage', jsonData);
+}
+
+function readData(): Entry[] {
+  const jsonEntries = localStorage.getItem('entries-storage');
+  if (!jsonEntries) {
+    return data.entries;
+  } else {
+    return JSON.parse(jsonEntries);
+  }
+}
```

Comment:
This function should be creating the whole `data` object, not just returning `data.entries`. Just like in the `typescript-local-storage` exercise, you will call `readData` and it will either pull `data` from localStorage or create the initial state of the object